utils/pubsub.go

```diff
@@ -14,6 +16,11 @@ type DataChannel chan DataEvent
 
 // DataChannelSlice is a slice of DataChannels
 type DataChannelSlice []DataChannel
+type EventBusI interface {
```

Comment:
Unlike in java, go interfaces only define behavior and hence it usually has some actionable names like Writer Reader etc.,
In this case it could be a PublishSubscriber.